screens/SendSetupScreen.js

```diff
@@ -295,7 +324,143 @@ const SendSetupScreen = ({
     }
   };
 
+  const handleDeepLink = ({
+    address,
+    amount,
+    tokenAmount,
+    label,
+    tokenId,
+    symbol
+  }): DeepLinkParams => {
+    const type = getType(address);
+
+    if (typeof type !== "string") {
+      setErrors(["Invalid Address"]);
+    }
+    if (type === "cashaddr") {
+      return prefillBCH({ address, amount, label });
+    }
+    if (type === "slpaddr") {
+      prefillSLP({
+        address,
+        amount,
+        tokenAmount,
+        label,
+        tokenId,
+        symbol
+      });
+    }
+  };
+
+  const prefillSLP = ({
+    address,
+    amount,
+    tokenAmount,
+    label,
+    tokenId,
+    symbol
+  }): DeepLinkParams => {
+    const hasTokenAmount = address !== undefined && tokenAmount !== undefined;
+
+    if (tokenId === undefined) {
+      setErrors(["Missing Token Type"]);
+      redirectHome(navigation);
+      return null;
+    }
+
+    if (hasTokenAmount) {
+      try {
+        setSendAmountCrypto(tokenAmount);
+        // const numericalAmount = parseAmount(tokenAmount);
+        setAmountType("crypto");
+        setSendAmount(tokenAmount);
+      } catch (error) {
+        setErrors(["Invalid Amount"]);
+        redirectHome(navigation);
+      }
+    }
+    setToAddress(address);
+  };
+
+  const prefillBCH = ({
+    address,
+    amount,
+    tokenAmount,
+    label,
+    tokenId,
+    symbol
+  }): DeepLinkParams => {
+    const hasAmount = address !== undefined && amount !== undefined;
+    navigation.state.params.symbol = "BCH";
+    if (hasAmount) {
+      try {
+        setSendAmountCrypto(amount);
+        const numericalAmount = parseAmount(amount);
+        setAmountType("crypto");
+        setSendAmount(amount);
+        setSendAmountFiat(
+          fiatRate
+            ? (fiatRate * (numericalAmount || 0)).toFixed(
+                currencyDecimalMap[fiatCurrency]
+              )
+            : 0
+        );
+      } catch (error) {
+        setErrors(["invalid amount"]);
+      }
+    }
+
+    setToAddress(address);
+  };
+
+  const parseQr = (
+    qrData: string
+  ): {
+    address: string,
+    amount: ?string,
+    tokenAmount: ?object,
+    label: ?string,
+    tokenId: ?string,
+    symbol: ?string
+  } => {
```

Comment:
This type makes no sense.
`qrData` should probably be of type `DeepLinkParams`?  Or something else?
But as right now, this says the `qrData` is a string, but also an object with a bunch of fields.